modules/drivers/presto-jdbc/src/metabase/driver/presto_jdbc.clj

```diff
@@ -250,36 +255,57 @@
               :subname     (mdb.spec/make-subname host port (db-name catalog schema))})
       prepare-addl-opts
       (dissoc :host :port :db :catalog :schema :tunnel-enabled :engine :kerberos)
-    sql-jdbc.common/handle-additional-options))
+      sql-jdbc.common/handle-additional-options))
 
 (defn- str->bool [v]
   (if (string? v)
     (Boolean/parseBoolean v)
     v))
 
+(defn- maybe-add-ssl-stores [details-map]
+  (let [props
+        (cond-> {}
+          (str->bool (:ssl-use-keystore details-map))
+          (assoc :SSLKeyStorePath (-> (secret/db-details-prop->secret-map details-map "ssl-keystore")
+                                      (secret/value->file! :presto-jdbc)
+                                      .getCanonicalPath)
+                 :SSLKeyStorePassword (secret/value->string
+                                       (secret/db-details-prop->secret-map details-map "ssl-keystore-password")))
+          (str->bool (:ssl-use-truststore details-map))
+          (assoc :SSLTrustStorePath (-> (secret/db-details-prop->secret-map details-map "ssl-truststore")
+                                        (secret/value->file! :presto-jdbc)
+                                        .getCanonicalPath)
```

Comment:
Should this be a `some->`? Wondering if the options could have `:ssl-use-truststore` but the secret resolves to nothing and the `.getCanonicalPath` will explode

Reply:
`some->` wouldn't help here, as no functions here can return `nil`, but indeed, `secret/value->file!` could throw a NPE if the property was not set. I'll add a specific check for this situation.